src/shogun/multiclass/KNNSolver.h

```diff
@@ -0,0 +1,114 @@
+/* This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Copyright (c) 2012-2013 Sergey Lisitsyn
+ */
+
+#ifndef KNNSOLVER_H__
+#define KNNSOLVER_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/common.h>
+#include <shogun/io/SGIO.h>
+#include <shogun/distance/Distance.h>
+#include <shogun/machine/DistanceMachine.h>
+
+namespace shogun
+{
+/**
+ * The KNNSolver class is the virtual base class of BruteKNNSolcer, CoverTreeKNNSolver, KDTreeKNNsolver, LSHKNNSolver. 
+ * The KNNSolver class will include most of Variables which been used all the other Solver and methods been used to choose 
```

Comment:
The KNNSolver class will include most of Variables which been used all the other Solver and methods been used to choose

In fact this sentence doesnt make any sense. Remove it and just say that this is the base class for all KNN solvers